Source/Blazorise/Components/Validation/EditContextValidator.cs

```diff
@@ -150,44 +150,48 @@ public virtual void ValidateField( EditContext editContext, ValidationMessageSto
         /// <returns>True if property is found.</returns>
         protected virtual bool TryGetValidatableProperty( in FieldIdentifier fieldIdentifier, out ValidationPropertyInfo validationPropertyInfo, bool forLocalization )
         {
-            var cacheKey = (ModelType: fieldIdentifier.Model.GetType(), fieldIdentifier.FieldName);
-
-            if ( !propertyInfoCache.TryGetValue( cacheKey, out validationPropertyInfo ) )
+            validationPropertyInfo = null;
+            if ( fieldIdentifier.FieldName is not null )
             {
-                // DataAnnotations only validates public properties, so that's all we'll look for
-                // If we can't find it, cache 'null' so we don't have to try again next time
-                var propertyInfo = cacheKey.ModelType.GetProperty( cacheKey.FieldName );
-
-                // This is used only for custom localization. We assume that unformatted ErrorMessage will be
-                // used as an localization key, so we need to replace it in case it is undefined with
-                // the internal ErrorMessageString that has unformatted message. eg. "The field {0} is invalid."
-                var validationAttributes = ValidationAttributeHelper.GetValidationAttributes( propertyInfo );
-                var formattedValidationAttributes = ValidationAttributeHelper.GetValidationAttributes( propertyInfo );
+                var cacheKey = (ModelType: fieldIdentifier.Model.GetType(), fieldIdentifier.FieldName);
 
-                if ( forLocalization )
+                if ( !propertyInfoCache.TryGetValue( cacheKey, out validationPropertyInfo ) )
```

Comment:
Hmm, I must test this. Caching might be tricky. There is a risk of getting the field reference to an older model instance.

Reply:
Tested and seems to be working fine.